### PR TITLE
Fix data and logging issues

### PR DESCRIPTION
**Pull request by jgillson**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -217,7 +217,7 @@ class Car
   def dealer_options(*args); new; end
   notifies :design, :safety
   # adding additional event payload options for any instance method(s), by passing a hash as the last argument
-  notifies :dealer_options, { sport_package: true }
+  notifies [:dealer_options, { sport_package: true }]
```
Comment (pboling) [inline README.md:220]: Is this change necessary?  If so, it would be a breaking change, and I'd like to avoid that.

File: spec/debug_logging/class_notifier_spec.rb [hunk 2]
```diff
@@ -20,10 +20,10 @@
         complete_notified_klass.k_with_dsplat(a: 'a')
       end
       expect(output).to match(/k_with_dsplat.log/)
-      expect(output).to match(/payload={:args=>\[{:a=>"a"}\]}/)
+      expect(output).to match(/payload={:debug_args=>\[{:a=>"a"}\]}/)
       expect(complete_notified_klass.instance_variable_get(DebugLogging::Configuration.config_pointer('k', :k_with_dsplat))).to receive(:log)
       expect(@events).to contain_exactly(
-        have_attributes(name: 'k_with_dsplat.log', payload: { args: [{ a: 'a' }] })
+        have_attributes(name: /k_with_dsplat.log/, payload: { debug_args: [{ a: 'a' }] })
```
Comment (pboling) [inline spec/debug_logging/class_notifier_spec.rb:26]: Prefer to use the string over the regex, because we want exact matches here.  If we can match on a more precise expectation we should.  Same goes for other string => regex updates elsewhere in specs.

File: spec/support/shared_context.rb [hunk 3]
```diff
@@ -107,7 +107,7 @@ def self.k_with_dsplat_payload(**_args)
         30
       end
       notifies :k_with_ssplat, :k_with_dsplat, :k_with_ssplat_error
-      notifies :k_with_dsplat_payload, { id: 1, first_name: 'Joe', last_name: 'Schmoe' }
+      notifies [:k_with_dsplat_payload, { id: 1, first_name: 'Joe', last_name: 'Schmoe' }]
```
Comment (pboling) [inline spec/support/shared_context.rb:110]: Again, hoping this change is not needed.

**Review by pboling — CHANGES_REQUESTED** [review]
A few minor changes

File: README.md (hunk 1 above)
Comment (jgillson) [reply README.md]: Actually it's not necessary - I had changed around the iteration logic slightly in `lib/debug_logging/class_notifier.rb` as I had thought it was necessary as part of the fix for the duplicate logging issue but it's not so I reverted it back to the original (similar to `/lib/debug_logging/class_logger.rb`).  I updated the `README.md` file accordingly.

File: spec/debug_logging/class_notifier_spec.rb (hunk 2 above)
Comment (jgillson) [reply spec/debug_logging/class_notifier_spec.rb]: I switched out all the matches on the regex with exact string matches throughout all the notification-related specs:

`spec/debug_logging/class_notifier_spec.rb`
`spec/debug_logging/instance_notifier_spec.rb`
`spec/debug_logging/log_subscriber_spec.rb`

File: spec/support/shared_context.rb (hunk 3 above)
Comment (jgillson) [reply spec/support/shared_context.rb]: I reverted this back as well as it's not needed - please see earlier [comment](https://github.com/pboling/debug_logging/pull/84#discussion_r516850231)

**Review by pboling — APPROVED** [review]
LGTM
